gen/flights/ai_flight_planner.py

```diff
@@ -433,6 +433,22 @@ def friendly_control_points(self) -> Iterator[ControlPoint]:
         return (c for c in self.game.theater.controlpoints if
                 c.is_friendly(self.is_player))
 
+    def farthest_friendly_control_point(self) -> ControlPoint:
+        """
+        Iterates over all friendly control points and find the one farthest away from the frontline
+        BUT! prefer Cvs. Everybody likes CVs!
+        """
+        fromFrontline = 0
+
+        for c in self.game.theater.controlpoints:
+            if c.is_carrier:
+                return c
+            if c.is_friendly(self.is_player) & c.has_frontline:
+                if c.distance_to(self.front_lines().__next__()) > fromFrontline:
+                    fromFrontline = c.distance_to(self.front_lines().__next__())
+                    cp = c
+        return cp
```

Comment:
Could ater `_targets_by_range` to be able to handle this:

```
    def _targets_by_range(
            self,
            targets: Iterable[MissionTarget],
            range_to: Optional[Iterable[MissionTarget]] = None) -> Iterator[MissionTarget]:
```

The test against the range to `self.friendly_cps()` would be replaced by `range_to`, and at the beginning of the function just do:

```
if range_to is None:
    range_to = self.friendly_cps()
```

Then this function is just:

````
for cp in reversed(self._targets_by_range(self.friendly_cps(), self.enemy_cps())):
    return cp
```

Reply:
If I append `if range_to is None:
    range_to = self.friendly_cps()` to the start of the function it doesnt work.
    
Moving this statement inside the ` for target in targets:` does kinda work, but then its throwing an error because i cannot reverse a generator

Reply:
Ah, I'd forgotten about the "but prefer over the CV" aspect. That wouldn't work with what I suggested.